x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/aws_credentials_form/aws_credentials_form.tsx

```diff
@@ -235,7 +235,9 @@ export const AwsCredentialsForm = ({
         size="m"
         options={getSetupFormatOptions()}
         idSelected={setupFormat}
-        onChange={onSetupFormatChange}
+        onChange={(idSelected: SetupFormat) =>
+          idSelected !== setupFormat && onSetupFormatChange(idSelected)
```

Comment:
btw we have almost identical code `x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx` but the bug is not present there, I guess due to the `setupFormat` and `onSetupFormatChange` not coming from a custom hook there. All that to say that maybe it makes sense to introduce the same change there so that we don't wonder later why it is handled differently in two places